\doxysection{exception/src/runtime/\+IOException.hpp File Reference}
\hypertarget{IOException_8hpp}{}\label{IOException_8hpp}\index{exception/src/runtime/IOException.hpp@{exception/src/runtime/IOException.hpp}}
{\ttfamily \#include "{}Runtime\+Exception.\+hpp"{}}\newline
{\ttfamily \#include $<$jpl/utils/debug/\+Error\+Utils.\+hpp$>$}\newline
\doxysubsubsection*{Data Structures}
\begin{DoxyCompactItemize}
\item 
class \mbox{\hyperlink{classjpl_1_1__exception_1_1IOException}{jpl\+::\+\_\+exception\+::\+IOException}}
\end{DoxyCompactItemize}
\doxysubsubsection*{Namespaces}
\begin{DoxyCompactItemize}
\item 
namespace \mbox{\hyperlink{namespacejpl}{jpl}}
\item 
namespace \mbox{\hyperlink{namespacejpl_1_1__exception}{jpl\+::\+\_\+exception}}
\end{DoxyCompactItemize}


\doxysubsection{Detailed Description}
An IOException is thrown when an error is occurred in runtime during an I/O operation either on a socket or a file

It is able to catch the last error occurred via \+\_\+utils\+::\+\_\+error\+::\+\_\+\+Get\+Last\+Error() if you instance via the default constructor; it even get the description of the error code via \+\_\+utils\+::\+\_\+error\+::\+\_\+\+Get\+Last\+Error\+As\+String(unsigned int)

\begin{DoxyDate}{Date}
2023-\/08-\/01 
\end{DoxyDate}
\begin{DoxyCopyright}{Copyright}
Copyright (c) 2023 
\end{DoxyCopyright}
\begin{DoxyAuthor}{Author}
Gabriele-\/\+P03 
\end{DoxyAuthor}
